imagesearch/inputs.py

```diff
@@ -0,0 +1,47 @@
+"""
+Data input pipeline definition.
+"""
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import division
+
+import tensorflow as tf
+
+def generate_input_fn(file_pattern, batch_size, num_epochs=None, shuffle=False):
+    "Return _input_fn for use with Experiment."
+
+    def _input_fn():
+        height, width, channels = [256, 256, 3]
+
+        filename_tensor = tf.train.match_filenames_once(file_pattern)
+        filename_queue = tf.train.string_input_producer(
+            filename_tensor,
+            num_epochs=num_epochs,
+            shuffle=shuffle)
+
+        reader = tf.WholeFileReader()
+        filename, contents = reader.read(filename_queue)
+
+        image = tf.image.decode_jpeg(contents, channels=channels)
+        image = tf.image.resize_image_with_crop_or_pad(image, height, width)
+        image_batch, filname_batch = tf.train.batch(
+            [image, filename],
+            batch_size,
+            num_threads=4,
+            capacity=50000)
+
+        image_batch = tf.to_float(image_batch) / 255
+        image_batch = (image_batch * 2) - 1
+
+        features = {
+            "image": image_batch,
+            "filename": filname_batch
```

Comment:
Yes. I'm using it at the very end with Annoy. So I can know which image to associate with which features.